mealie/schema/response/pagination.py

```diff
@@ -31,6 +32,7 @@ class PaginationQuery(MealieModel):
     order_by: str = "created_at"
     order_direction: OrderDirection = OrderDirection.desc
     query_filter: str | None = None
+    timestamp: float | None
```

Comment:
I think using a string and defaulting to empty would help avoid some possible null errors. Maybe be worth renaming this to `randomSeed` for clarity as well

```suggestion
    timestamp: string = ""
```

Reply:
My thinking here is that timestamps can be used as seeds, but seeds cannot be used as timestamps. So months down the road, if someone wants to do something time between searches, they could easily see this variable name and know that it contains the timestamp. Would that make sense?

For turning it into a string, that would then make things more complicated. random.seed() takes NoneType values, so if an empty string somehow made it past validation that would be an Exception. But if the current implementation somehow sneaks past unset, then random.seed() still works (though it would start resetting the seed every page). So to me, it would make sense to leave it as a float/None. What do you think?

Reply:
> My thinking here is that timestamps can be used as seeds, but seeds cannot be used as timestamps. So months down the road, if someone wants to do something else with it (time between searches?), they could easily see this variable name and know that it contains the timestamp. Would that make sense?

But there's no validation that is in a timestamp, so it's really just any float to be used as a seed. 

> so if an empty string somehow made it past validation that would be an Exception.

I don't think that's a reasons to adjust programming. You can easily test that empty strings are validated correctly.

> But if the current implementation somehow sneaks past unset, then random.seed() still works (though it would reset the seed every page). And I think of a time stamp as an int/float but a date/time as a string.

If the users input is invalid, like in the case of `order_by = "random"` and `timestamp = ""` then we should return the user an error so they can use the API correctly. We shouldn't accept malformed data and _try_ to do something with it. That always leads to confusion on the consumer side.

Reply:
True that there’s no validation it’s a time stamp if you provide it via the API. Which is why I was setting it to now() if it wasn’t provided. 

I don’t really feel strongly about the future-proofing I mentioned, so I’m happy to just rename the variable, pass as a string to random, and return an invalid error if random is set but randomSeed is not.